receiver/prometheusexec/subprocessmanager/manager.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subprocessmanager
```

Comment:
Would you want anything in this package to be exported as part of the external interface to your receiver? Consider moving it into an `internal` package instead.

Reply:
I did see some value in leaving it as its own package in case someone could use it in the future for something similar to my receiver but with another service than `Prometheus`. On its own, it starts processes, pipes their output to `zap.Logger`, computes exponential backoff and keeps track of runtime. Do you see any value in this, if not I can for sure make it part of an `internal` package.